### ISSUE TITLE
Export Crashes Blender

### ISSUE THREAD

**Issue by lyonheart001** [AsoboStudio/glTF-Blender-IO-MSFS#141]
### Current Behavior

Hello Team,

I exported Blender 3.3 this morning, unknowing it was a new build version 'series'.  I did the usual install of your SDK to Blender, tried to export my plane with it as a test and it would crash Blender, CTD.

I then saw this is a totally new version of Blender.  So I went back to Blender 3.2 to see if it would export, to clean up the exported glTF fail.  And 3.2 also crashed to desktop on the export.

I have now reverted back to 3.2 May 27th Build, and that too no longer exports.  No way to find out as it CTD's.  No messages.

It might just be my model, as the exterior exports fine, but now the interior is not exporting.  If only there was a way to do a check on the model to see what is causing this.

Bill
Lionheart Creations

### Expected Behavior

Export to glTF 2 for MSFS

### Steps To Reproduce

Select all parts in Interior scene
Export in Asobo glTF2 format for MSFS

### Environment

```markdown
- OS: WIN 10
- Blender: 3.2 APR 30th, APR 28th, APR 27th, and also new Blender 3.3 May 2nd build.
- glTF-Blender-IO-MSFS: Latest build from 6 days ago, and also tested 2 builds back to see if that worked on Blender 3.3.  It did not.
```

### Anything else?

To my knowledge and memory, it was exporting fine.  I know of nothing that has changed.  I also tried reverting back to earlier settings, nothing worked.  I also used a pure version saved blender file that Blender 3.3 hadnt save to export from Blender 3.2, but still crashes.

**Comment by btronquo** [comment]
Hi,

First, I suggest you to make a clean install (remove all custom addons you installed too) of blender 3.1 (not 3.2 or 3.something) 

Then, install the plugin and try again

**Comment by lyonheart001** [comment]
Ok, I did some cleaning.  I also tried the install path with dropping the folder into the Windows core routing and that finally got it to show up.  
Two notes;
1. The gltf folder has the same name as the kit.  I was trying to install the entire pack instead of the Addons/gltf folder which is inside it.  That was one mistake I was doing.  I humbly suggest renaming it, like SDK_glTF_Exporter_Kit or something, so people know to look in there for the Addon/gltf.....  folder which needs to be installed.
2.  When it is now in Blender, it looks like the original 'Blender' glTF system, not the older version that said it was an MSFS exporter, so I had thought it wasnt showing up when in reality it was.  It was appearing like the original Blender basic version glTF import/export system.

So... Its now running, I have the SDK in place once again.   It works in the new Blender 3.3 as well as 3.2, and my model does indeed have an issue with its 'interior' model exporting.  Something happened in the past several days, evidently, to break it, causing it to crash Blender when exporting.  But the exterior side of the plane does export without crashing Blender 3.2 or Blender 3.3 and shows up in the aircraft model folder as exported.  (Not tested in MSFS yet).

Hope that helps.

Bill / LHC

**Comment by lyonheart001** [comment]
Question..  How can one see what is causing Blender to crash when doing an export?

**Comment by btronquo** [comment]
I suggest you to search on google ;)
I found this :
https://docs.blender.org/manual/en/latest/troubleshooting/crash.html

> When Blender crashes, it writes out a text file which contains information that may help identify the cause of the crash. Usually, this file is written in the [Temporary Directory](https://docs.blender.org/manual/en/latest/advanced/blender_directory_layout.html#temp-dir) directory.
> 
> This file contains a log of tools used up until the crash as well as some other debug information. When reporting bugs about crashes it can be helpful to attach this file to your reports, especially when others are unable to reproduce the crash.

**Comment by lyonheart001** [comment]
I never found crash logs.  I looked all over both drives.

I also recleaned my addons 'again' and that broke the SDK.  You need the basic Blender glTF import/export addon that comes with Blender, as I think the Asobo SDK uses this.  When I got that back, the SDK was good again.

what a mess...  :(   At least its exporting the exterior side of the model again.

I think the crashes are a memory issue, somehow.  I did an increase to systems cache for Blender, put it on a diff drive (cache folder) and raised that to 200 GB, and now, when the SDK is exporting the interior, it just continually runs instead of crashes.  Odd.

I have to redo my Materials again...  Losing the SDK must reset the 'Migration' system.

Bill Ortis
Lionheart Creations

**Comment by lyonheart001** [comment]
Goodness, some crazy things have happened.
* Base top layers of many materials are missing, and they are also missing from my Textures folder, like they were deleted.
* Going back to an old Blender 3.1 installation, opening my plane, all files were fine.  I thought this might be Ron's older updated base MSFS installer that used Vitus' work, but no, it was an earlier version of Asobo SDK exporter, Build 1.7.5.
I tried running the latest Build of the Asobo for 3 days, Build 1.1.6.  
I do not understand how it goes from 1.7.5 to 1.1.6 ?

Also, in Build 1.7.5 in Blender 3.1, all the materials are fine...   now they are mostly all messed up and have a 'Migrate' button on them, which I was going through trying to fix them all.  It occurred to me to try an early version of Blender to see what the textures did, and sure enough, they were fine, and this was with a version of the Asobo exporter.

I dont understand what has happened.  But I think for now, I will be running 1.7.5.

**Comment by lyonheart001** [comment]
I thought Asobo's SDK was runnning perfectly fine on Blender 3.2?

**Comment by lpierabella** [comment]
@lyonheart001 did you solve your problem? can we close this issue? 
In case your error comes from a misunderstand of the readme I invite you to make a pull request adding the missing details in the readme

**Comment by lyonheart001** [comment]
> @lyonheart001 did you solve your problem? can we close this issue? In case your error comes from a misunderstand of the readme I invite you to make a pull request adding the missing details in the readme

Hello Ipierablella,
I believe I know what happened.  I think I was still running the RonH exporter, accidentally installed the Asobo version into a new Blender, and it made a mess of my materials.  I remember I was holding off on the Asobo version until my plane project was done.  I messed up.  

This might explain the crashes.  The materials might be causing this.  I have been repairing them one by one.

My humble apologies.  Yes, please close it.

Kind regards,
Bill
LHC

**Comment by lyonheart001** [comment]
Ipierabella, 

I have a quick question.  I have been unable to export my interior portion of my model for many days now.  I click on the export, and it runs for a moment and then Blender CTD's.  I have not been able to find a crash note to figure out why its crashing.  I was able to see what the memory useage and chip useage amounts were when I clicked on Export; both at about 30%, which should be ok, I would think.  It was working fine just a few days ago.  What would you do to try to figure out what is going wrong, why its crashing?

Kind regards,

Bill
LHC

**Comment by lpierabella** [comment]
there is no blender.crash.txt in your %TEMP% folder?

**Comment by lyonheart001** [comment]
> there is no blender.crash.txt in your %TEMP% folder?

None.

Last night, I loaded a fresh install of Blender 3.2 and Asobo SDK on my new Laptop.  Pure drive, Windows 11, never had Blender on it.  Blender crashed on that export also.  A week ago, it was fine on my main work computer.

I wonder what has caused this?  

At one point, I did get this report I think on Blender 3.2, where it refused to export and showed this crash window;

![Screenshot 2022-05-04 164854](https://user-images.githubusercontent.com/76859194/167163686-e006d0ac-e413-46bd-bfdc-7d5e578e441d.jpg)

**Comment by lyonheart001** [comment]
Also...  I did find that with Blender 3.1 and Rons SDK exporter, my project exports 'perfect'.  Same Blender scene file from last week , old Blender 3.1, and Rons SDK worked, and materials were fine.  

With 3.2 and 3.3, its like base materials were gone (textures) and various other odd issues.

It could be Windows changed...    I do not think it could be MSFS SU9.  How could it effect exporting of a Blender file in glTF format?  I updated my nVidia driver 2 days ago to see if that would fix it.

So odd...
